### ISSUE TITLE
Multi-coloured GraphViz lines (Origin: bugzilla #694870)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#5103]
status NEW severity *enhancement* in component *general* for *---*
Reported in version *1.8.3.1* on platform *Other*
Assigned to: Dimitri van Heesch

On 2013-02-28 14:29:08 +0000, David King wrote:

    Not sure if this is one for Doxygen, or one for GraphViz, or both.
    But could we have the option of multicoloured arrow lines, in GraphViz (dot)
    called and caller graph: For a given arrow line, one of 20 or so easily
    distinquishable colours to be randomly chosen. Just so can more easily
    distinguish arrow lines on complex graphs with overlapping lines. Could be
    enhanced by mnaking a) random colour selection beikng subject to disallowing
    same colour for two adjacent line starts around a box. Ditto line ends. Or
    perhaps b) disallowing any two same colours around a box. Inless palette of
    20 exhausted, in which case allow rule b) relax.

**Comment by albert-github** [comment]
I think having a multitude of colored lines gives a very messy impression and with complex images the result is still very hard to follow.
In the current doxygen, 1.9.8, version the SVG version of the images have in the interactive mode the feature what when hovering over a line this line is highlighted and when hovering over a box / node its parents and children are highlighted.

I think the problem has been solved in the mean time.


Note the "Class Hierarchy" graphs don't use this feature as their readability is normally OK. The disabling of the feature for these diagrams is done in dotgfxhierarchytable.cpp in the function `DotGfxHierarchyTable::createGraph` where we see `m_zoomable = FALSE;`.
